cloudera/datadog_checks/cloudera/api_client_v7.py

```diff
@@ -22,118 +21,81 @@ def _collect_clusters(self):
-
     def _collect_cluster_metrics(self, cluster_name, tags):
-        time_series_resource_api = cm_client.TimeSeriesResourceApi(self._api_client)
         metric_names = ','.join(f'last({metric})' for metric in TIMESERIES_METRICS['cluster'])
         query = f'SELECT {metric_names} WHERE clusterName="{cluster_name}" AND category=CLUSTER'
-        self._log.debug('query: %s', query)
-        query_time_series_response = time_series_resource_api.query_time_series(query=query)
-        self._log.debug("Full timeseries response:")
-        self._log.debug(query_time_series_response)
-        self._collect_query_time_series(query_time_series_response, 'cluster', [])
+        self._query_time_series(query, category='cluster', tags=tags)
 
-    def _collect_hosts_metrics_and_service_check(self, cluster_name):
+    def _collect_hosts(self, cluster_name):
         clusters_resource_api = cm_client.ClustersResourceApi(self._api_client)
         list_hosts_response = clusters_resource_api.list_hosts(cluster_name, view='full')
         self._log.debug("Full hosts response:")
         self._log.debug(list_hosts_response)
-
         for host in list_hosts_response.items:
-            self._collect_host_metrics_and_service_check(host)
-
-    def _collect_host_metrics_and_service_check(self, host):
-        self._log.debug('host: %s', host)
+            tags = [
+                f'cloudera_hostname:{host.hostname}',
+                f'cloudera_rack_id:{host.rack_id}',
+                f'cloudera_host_id:{host.host_id}',
+                f'cloudera_cluster:{host.cluster_ref.cluster_name}',
+            ]
+
+            host_tags = host.tags
+            if host_tags:
+                for host_tag in host_tags:
+                    tags.append(f"{host_tag.name}:{host_tag.value}")
+
+            if host.host_id:
+                self._collect_host_metrics(host, tags)
+                self._collect_host_service_check(host, tags)
+
+    def _collect_host_service_check(self, host, tags):
         host_entity_status = ENTITY_STATUS[host.entity_status] if host.entity_status else None
-
-        host_id = host.host_id
         self._log.debug('host_entity_status: %s', host_entity_status)
-        self._log.debug('host_id: %s', host_id)
+        self._check.service_check(HOST_HEALTH, host_entity_status, tags=tags)
 
-        if host_id:
-            tags = [f"{tag.name}:{tag.value}" for tag in host.tags] if host.tags else []
-            self._collect_host_metrics(host_id, tags)
-            self._collect_role_metrics(host_id, tags)
-            self._collect_host_disks(host_id)
-            self._log.debug('host tags: %s', tags)
-            self._check.service_check(HOST_HEALTH, host_entity_status, tags=tags)
+    def _collect_host_metrics(self, host, tags):
+        categories = TIMESERIES_METRICS.keys()
+        for category in categories:
+            metric_names = ','.join(f'last({metric})' for metric in TIMESERIES_METRICS[category])
```

Comment:
Are we getting all the categories now instead of only host?

Reply:
After talking offline, this will be changed back to individual functions rather than combining all of them.